### DISCUSSION TITLE
Calculated fields not working after upgrading to Pimcore X

### DISCUSSION THREAD

**Discussion by Hbitvof** [pimcore/pimcore#12059] [General]
Hi,

All calculated fields made in Pimcore 6.x stopped working when I upgrade to Pimcore X.

In the documentation nothing is mentioned about this. Any suggetion what I need to change for this part?

The only thing I changed so far is the calculator class path in Pimcore and the namespace path because this changed in Pimcore X.

**Comment by dvesh3** [comment]
AFAIK nothing really changed in regards to calculated fields on Pimcore X. Could you please provide some information on implementation? thanks!

**Comment by blankse** [comment]
Do you use the CalculatorClassInterface? Since Pimcore X this is required. The BC layer was removed. See https://github.com/pimcore/pimcore/pull/7446

**Reply by dvesh3** [reply]
@blankse I wonder why this is not in upgrade notes 😞

**Reply by Hbitvof** [reply]
@blankse I don't use the CalculatorClassInterface. Btw, why is this still being used in the demo files (for example https://github.com/pimcore/demo/blob/10.x/src/Model/Product/CalculatedValue/AccessoryPartName.php)?

**Reply by dvesh3** [reply]
@Hbitvof Sebastian meant that you **must** implement `CalculatorClassInterface` for calculator class to work on Pimcore X.

**Reply by blankse** [reply]
@dvesh3 I created a PR for the upgrade notes https://github.com/pimcore/pimcore/pull/12100

**Reply by dvesh3** [reply]
@blankse you're a lifesaver! 💜

**Comment by Hbitvof** [comment]
@dvesh3 and @blankse, thanks for the response.

I try to give some details I did. When I am at Pimcore 6.x, the path to the calculator class in PIM is \AppBundle\Model\Product\CalculatedValue\Classname

After migration I tested with path `@App\Model\Product\CalculatedValue\Classname` or `\App\Model\Product\CalculatedValue\Classname`.

Within the class file itself I changed `namespace AppBundle\Model\Product\CalculatedValue;` to `namespace App\Model\Product\CalculatedValue;`

As test I also made a new calculated value that is always returing the text "test" but the calculated value is empty.

A reference I looked at the demo files on Github and the demo PIMCore website but still the calculated value is empty.

I have a feeling something is going wrong with the path.

**Reply by dvesh3** [reply]
It works on [demo.pimcore.fun](https://demo.pimcore.fun/admin):
![image](https://user-images.githubusercontent.com/5137917/167572427-a318050f-e5c7-4512-a985-c86e994e36c9.png)

![image](https://user-images.githubusercontent.com/5137917/167572324-4fad61f0-ee29-4b15-b6b3-d65792bc3d78.png)

**Comment by Hbitvof** [comment]
I got it working now, thanks for your help. The problem was the implementation of the CalculatorClassInterface for calculator class to work on Pimcore X.
